library/core/src/task/wake.rs

```diff
@@ -33,9 +35,18 @@ impl RawWaker {
     /// The value of this pointer will get passed to all functions that are part
     /// of the `vtable` as the first parameter.
     ///
+    /// It is important to consider that the `data` pointer must point to a
+    /// thread safe type such as an `[Arc]<T: Send + Sync>`
+    /// when used to construct a [`Waker`]. This restriction is lifted when
+    /// constructing a [`LocalWaker`], which allows using types that do not implement
+    /// <code>[Send] + [Sync]</code> like `[Rc]<T: !Send + !Sync>`.
```

Comment:
`Rc<T>` is never Send/Sync, regardless of the T within, so this description feels a bit off to me. Maybe just drop those bounds here?